src/solvers/flattening/arrays.h

```diff
@@ -77,6 +82,27 @@ class arrayst:public equalityt
   typedef std::map<std::size_t, index_sett> index_mapt;
   index_mapt index_map;
 
+  enum class constraint_typet
+  {
+    ARRAY_ACKERMANN,
+    ARRAY_EQUALITY,
+    ARRAY_WITH,
+    ARRAY_WITH_OTHER,
+    ARRAY_IF_TRUE,
+    ARRAY_IF_FALSE,
+    ARRAY_OF,
+    ARRAY_TYPECAST,
+    ARRAY_CONSTANT,
+    ARRAY_NON_CONSTANT,
+    ARRAY_COMPREHENSION
+  };
+
+  typedef std::list<exprt> array_constraintst;
+  typedef std::map<constraint_typet, array_constraintst> array_constraints_mapt;
+  array_constraints_mapt array_constraints_map;
```

Comment:
❗Storing copies these constraints is not ideal wrt memory consumption.

Reply:
No longer applicable, changed to store only number of constraints, don't store the constraints themselves.